internal/app/backend/backend_service_test.go

```diff
@@ -168,6 +168,132 @@ func TestDoFetchMatchesFilterChannel(t *testing.T) {
 	}
 }
 
+func TestDoReleaseTickets(t *testing.T) {
+	fakeProperty := "test-property"
+	fakeTickets := []*pb.Ticket{
+		{
+			Id: "1",
+			SearchFields: &pb.SearchFields{
+				DoubleArgs: map[string]float64{
+					fakeProperty: 1,
+				},
+			},
+		},
+		{
+			Id: "2",
+			SearchFields: &pb.SearchFields{
+				DoubleArgs: map[string]float64{
+					fakeProperty: 2,
+				},
+			},
+		},
+		{
+			Id: "3",
+			SearchFields: &pb.SearchFields{
+				DoubleArgs: map[string]float64{
+					fakeProperty: 2,
+				},
+			},
+		},
+	}
+
+	tests := []struct {
+		description   string
+		preAction     func(context.Context, context.CancelFunc, statestore.Service, *pb.Pool)
+		req           *pb.ReleaseTicketsRequest
+		wantCode      codes.Code
+		pool          *pb.Pool
+		expectTickets []string
+	}{
+		{
+			description: "expect unavailable code since context is canceled before being called",
+			preAction: func(_ context.Context, cancel context.CancelFunc, _ statestore.Service, pool *pb.Pool) {
+				cancel()
+			},
+			req: &pb.ReleaseTicketsRequest{
+				TicketIds: []string{"1"},
+			},
+			wantCode: codes.Unavailable,
+		},
+		{
+			description: "expect ok code when submitted list is empty",
+			preAction: func(_ context.Context, cancel context.CancelFunc, _ statestore.Service, pool *pb.Pool) {
+				cancel()
```

Comment:
My gut tells me this test is failing because the context is cancelled before it is used ;)

Reply:
Ah! So silly. I'm a bit new to reading some of these lambda-setup test patterns. Thanks! I'll fix it up